pkg/controllers/mcs/serviceexport_controller.go

```diff
@@ -138,6 +138,28 @@ func (c *SeController) Handle(obj interface{}) (requeueAfter *time.Duration, err
 		return nil, err
 	}
 
+	// remove endpoint slices exist in parent but not in child cluster
+	localEndpointSliceMap := make(map[string]bool, 0)
+	for _, item := range endpointSliceList {
+		localEndpointSliceMap[fmt.Sprintf("%s-%s", se.Namespace, item.Name)] = true
+	}
+	if parentEndpointSliceList, err := c.parentk8sClient.DiscoveryV1().EndpointSlices(c.dedicatedNamespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{discoveryv1.LabelServiceName: utils.DerivedName(namespace, seName)}).String()}); err == nil {
+		shouldRetry := false
+		for _, item := range parentEndpointSliceList.Items {
+			if !localEndpointSliceMap[item.Name] {
```

Comment:
This may not be right, since L144 uses namespaced name as keys.

Reply:
On L144, the enpointslice `item` is in child cluster, and the name is raw enpointslice name, like `my-service-pt9nk`. 

While here, L149, the item is in parent cluster enpointslice, the name is `default-my-service-pt9nk` formated on L270.

Reply:
> This may not be right, since L144 uses namespaced name as keys.

On L144, the enpointslice item is in child cluster, and the name is raw enpointslice name, like my-service-pt9nk.

While here, L149, the item is in parent cluster enpointslice, the name is default-my-service-pt9nk formated on L270.

Reply:
Ok, I got it.

This is a little bit tricky. Maybe we should add some comments here. Feel free to improve this in a new PR.